### PR TITLE
Bschnurr/unittest ui changes

### PR DESCRIPTION
**Pull request by bschnurr**

### CODE REVIEW

File: Python/Product/BuildTasks/Microsoft.PythonTools.targets [hunk 1]
```diff
@@ -205,12 +205,6 @@ permissions and limitations under the License.
     </CreatePythonCommandItem>
   </Target>
 
-  <PropertyGroup>
-    <PyTestEnabled>false</PyTestEnabled>
-    <PyTestPath>pytest</PyTestPath>
-    <PyTestArgs></PyTestArgs>
-  </PropertyGroup>
```
Comment (bschnurr) [inline Python/Product/BuildTasks/Microsoft.PythonTools.targets]: removing default properties.

File: Python/Product/PythonTools/PythonConstants.cs [hunk 2]
```diff
@@ -128,12 +128,13 @@ static class PythonConstants {
...
-        public const string PyTestPathSetting = "PyTestPath";
-        public const string PyTestArgsSetting = "PyTestArgs";
-        public const string UnitTestArgsSetting = "UnitTestArgs";
         public const string TestFrameworkSetting = "TestFramework";
 
+        public const string UnitTestRootDirectorySetting = "UnitTestRootDirectory";
+        public const string DefaultUnitTestRootDirectory = ".";
+        public const string UnitTestPatternSetting = "UnitTestPattern";
+        public const string DefaultUnitTestPattern = "test*.py";
+
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonConstants.cs]: new project properties and defaults for unittest

File: Python/Product/TestAdapter.Executor/Config/PythonProjectSettings.cs [hunk 3]
```diff
@@ -36,7 +36,9 @@ public class PythonProjectSettings : IEquatable<PythonProjectSettings> {
             IsWorkspace = isWorkspace;
             UseLegacyDebugger = useLegacyDebugger;
             TestFramwork = TestFrameworkType.None;
-            Enum.TryParse<TestFrameworkType>(testFramework, ignoreCase:true, out TestFramwork);
+            Enum.TryParse<TestFrameworkType>(testFramework, ignoreCase: true, out TestFramwork);
+            UnitTestPattern = string.IsNullOrEmpty(unitTestPattern) ? PythonConstants.DefaultUnitTestPattern : unitTestPattern;
+            UnitTestRootDir = string.IsNullOrEmpty(unitTestRootDir) ? PythonConstants.DefaultUnitTestRootDirectory : unitTestRootDir;
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Config/PythonProjectSettings.cs]: in the out of proc side we fallback to defaults if nothing was passed in

File: Python/Product/TestAdapter/PythonRunSettings.cs [hunk 4]
```diff
@@ -186,6 +188,8 @@ ref outObj
                             writer.WriteAttributeString("workingDir", config.WorkingDirectory);
                             writer.WriteAttributeString("interpreter", config.GetInterpreterPath());
                             writer.WriteAttributeString("pathEnv", config.Interpreter.PathEnvironmentVariable);
+                            writer.WriteAttributeString("unitTestRootDir", unitTestRootDir);
+                            writer.WriteAttributeString("unitTestPattern", unitTestPattern);
```
Comment (bschnurr) [inline Python/Product/TestAdapter/PythonRunSettings.cs]: when writing out the unittest properties to runtimesettings we dont try to validate and use defaults here.

File: Python/Product/PythonTools/PythonTools/Project/PythonTestPropertyPageControl.cs [hunk 5]
```diff
@@ -40,19 +38,25 @@ public partial class PythonTestPropertyPageControl : UserControl {
...
+            UnitTestRootDir = string.IsNullOrEmpty(rootDir) ? PythonConstants.DefaultUnitTestRootDirectory : rootDir;
+
+            string pattern = _propPage.Project.GetProjectProperty(PythonConstants.UnitTestPatternSetting, false);
+            UnitTestPattern = string.IsNullOrEmpty(pattern) ? PythonConstants.DefaultUnitTestPattern : pattern;
+
             string testFrameworkStr = _propPage.Project.GetProjectProperty(PythonConstants.TestFrameworkSetting, false);
-            TestFramework = TestFrameworkType.None; 
-            if(Enum.TryParse<TestFrameworkType>(testFrameworkStr, ignoreCase:true, out TestFrameworkType parsedFramworked)) {
+            TestFramework = TestFrameworkType.None;
+            if (Enum.TryParse<TestFrameworkType>(testFrameworkStr, ignoreCase: true, out TestFrameworkType parsedFramworked)) {
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/Project/PythonTestPropertyPageControl.cs]: `parsedFramework`

**Pull request merged**